pkg/model/pki.go

```diff
@@ -30,65 +32,80 @@ type PKIModelBuilder struct {
 
 var _ fi.ModelBuilder = &PKIModelBuilder{}
 
+// Build is responsible for generating the pki assets for the cluster
 func (b *PKIModelBuilder) Build(c *fi.ModelBuilderContext) error {
 	{
-		// Keypair used by the kubelet
 		t := &fitasks.Keypair{
 			Name:      fi.String("kubelet"),
 			Lifecycle: b.Lifecycle,
-
-			Subject: "o=" + user.NodesGroup + ",cn=kubelet",
-			Type:    "client",
+			Subject:   "o=" + user.NodesGroup + ",cn=kubelet",
+			Type:      "client",
 		}
 		c.AddTask(t)
 	}
 
 	{
-		// Secret used by the kubelet
-		// TODO: Can this be removed... at least from 1.6 on?
-		t := &fitasks.Secret{
-			Name:      fi.String("kubelet"),
+		t := &fitasks.Keypair{
+			Name:      fi.String("kube-scheduler"),
 			Lifecycle: b.Lifecycle,
+			Subject:   "cn=" + user.KubeScheduler,
+			Type:      "client",
 		}
 		c.AddTask(t)
 	}
 
 	{
-		// Keypair used by the kube-scheduler
 		t := &fitasks.Keypair{
-			Name:      fi.String("kube-scheduler"),
+			Name:      fi.String("kube-proxy"),
 			Lifecycle: b.Lifecycle,
-
-			Subject: "cn=" + user.KubeScheduler,
-			Type:    "client",
+			Subject:   "cn=" + user.KubeProxy,
+			Type:      "client",
 		}
 		c.AddTask(t)
 	}
 
 	{
-		// Secret used by the kube-scheduler
-		// TODO: Can this be removed... at least from 1.6 on?
-		t := &fitasks.Secret{
-			Name:      fi.String("system:scheduler"),
+		t := &fitasks.Keypair{
+			Name:      fi.String("kube-controller-manager"),
 			Lifecycle: b.Lifecycle,
+			Subject:   "cn=" + user.KubeControllerManager,
+			Type:      "client",
 		}
 		c.AddTask(t)
 	}
 
-	{
-		// Keypair used by the kube-proxy
-		t := &fitasks.Keypair{
-			Name:      fi.String("kube-proxy"),
-			Lifecycle: b.Lifecycle,
-
-			Subject: "cn=" + user.KubeProxy,
-			Type:    "client",
+	// @check if we need to generate certificates for etcd peers certificates from a different CA
+	// @question i think we should use another KeyStore for this, perhaps registering a EtcdKeyStore given
+	// that mutual tls used to verify between the peers we don't was
```

Comment:
Sorry, the comment was only half finished, was probably context switching at the time :-) .. But you got the gist. The outstanding question being whether we should be using a different CA for the etcd peers given auth is based purely on CA signing (would have been nice is they added a common name check at a minimum, but hey) ... 

I wasn't sure on a solution really! .. An intermediate ca would work but i'm guessing would require more work than adding another keystore. I think it would be enough to say, different CA for peers, block of peering port, change calico to use either kubernetes or worst case a client cert locked down to a specific etcd role post v3.